include/oneapi/tbb/blocked_rangeNd.h

```diff
@@ -110,20 +110,21 @@ class blocked_rangeNd_impl<Value, N, detail::index_sequence<Is...>> {
     static_assert(N != 0, "zero dimensional blocked_rangeNd can't be constructed");
 
     //! Ranges in each dimension.
-    std::array<tbb::blocked_range<value_type>, N> my_dims;
+    std::array<dim_range_type, N> my_dims;
 
     template<typename split_type>
     void do_split(blocked_rangeNd_impl& r, split_type proportion) {
-        static_assert((std::is_same<split_type, split>::value || std::is_same<split_type, proportional_split>::value), "type of split object is incorrect");
+        static_assert((std::is_same<split_type, split>::value || std::is_same<split_type, proportional_split>::value),
+                      "type of split object is incorrect");
         __TBB_ASSERT(r.is_divisible(), "can't split not divisible range");
 
-        auto my_it = std::max_element(my_dims.begin(), my_dims.end(), [](const tbb::blocked_range<value_type>& first, const tbb::blocked_range<value_type>& second) {
-            return (first.size() * second.grainsize() < second.size() * first.grainsize());
+        auto my_it = std::max_element(my_dims.begin(), my_dims.end(), [](const dim_range_type& first, const dim_range_type& second) {
+            return (first.size() * double(second.grainsize()) < second.size() * double(first.grainsize()));
```

Comment:
Why cast to double?

Reply:
That's what all other blocked_range implementations do :) I guess it is to avoid the risk of multiplication overflow.

Reply:
Ohh, I see